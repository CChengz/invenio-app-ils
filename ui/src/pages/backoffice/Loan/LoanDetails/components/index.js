export { LoanActions } from './LoanActions';
export { LoanMetadata } from './LoanMetadata';
export { AvailableItems } from './AvailableItems';
