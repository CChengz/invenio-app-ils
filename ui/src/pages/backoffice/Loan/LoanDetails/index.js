import { connect } from 'react-redux';

import { fetchLoanDetails } from './state/actions';
import LoanDetailsComponent from './LoanDetails';

const mapDispatchToProps = dispatch => ({
  fetchLoanDetails: loanPid => dispatch(fetchLoanDetails(loanPid)),
});

export const LoanDetails = connect(
  null,
  mapDispatchToProps
)(LoanDetailsComponent);
