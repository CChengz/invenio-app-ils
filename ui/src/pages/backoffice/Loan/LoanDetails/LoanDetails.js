import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { Loader, Error } from '@components';
import { LoanMetadata } from './components';

export default class LoanDetails extends Component {
  constructor(props) {
    super(props);
    this.fetchLoanDetails = this.props.fetchLoanDetails;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.loanPid) {
        this.fetchLoanDetails(location.state.loanPid);
      }
    });
    this.fetchLoanDetails(this.props.match.params.loanPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    const { isLoading, error } = this.props;
    return (
      <Container>
        <Loader isLoading={isLoading}>
          <Error error={error}>
            <Container>
              <LoanMetadata />
            </Container>
          </Error>
        </Loader>
      </Container>
    );
  }
}

LoanDetails.propTypes = {
  fetchLoanDetails: PropTypes.func.isRequired,
};
