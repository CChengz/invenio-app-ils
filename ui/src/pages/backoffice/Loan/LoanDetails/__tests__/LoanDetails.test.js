import React from 'react';
import { shallow, mount } from 'enzyme';
import LoanDetails from '../LoanDetails';

jest.mock('../components/', () => {
  return {
    LoanMetadata: () => null,
  };
});

describe('LoanDetails tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      loanPid: 879,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <LoanDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchLoanDetails={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch loan details on mount', () => {
    const mockedFetchLoanDetails = jest.fn();
    component = mount(
      <LoanDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchLoanDetails={mockedFetchLoanDetails}
      />
    );
    expect(mockedFetchLoanDetails).toHaveBeenCalledWith(
      routerUrlParams.params.loanPid
    );
  });
});
