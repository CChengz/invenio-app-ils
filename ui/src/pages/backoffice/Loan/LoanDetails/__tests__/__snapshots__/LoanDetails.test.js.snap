// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanDetails tests should load the details component 1`] = `
<Container>
  <Loader>
    <Error>
      <Container>
        <LoanMetadata />
      </Container>
    </Error>
  </Loader>
</Container>
`;
